app/lib/actions/katello/repository/filtered_index_content.rb

```diff
@@ -23,11 +23,11 @@ def run
             unit_ids = search_units(repo)
             ::Katello::Deb.import_all(unit_ids, repo)
           elsif repo.yum?
-            unit_ids = search_units(repo)
             if input[:content_type] == 'srpm'
+              unit_ids = search_units(repo)
               ::Katello::Srpm.import_all(unit_ids, repo)
             else
-              ::Katello::Rpm.import_all(unit_ids, repo)
+              ::Katello::Rpm.import_for_repository(repo)
```

Comment:
only passing in unit_ids was an optimization so that the entire repository doesn't get indexed.  This can be very slow on a large repository.